src/box/lua/schema.lua

```diff
@@ -3427,6 +3432,55 @@ box.schema.user.exists = function(name)
     end
 end
 
+-- Return normalized origins from the given tuple.
+local function origins_from_tuple(tuple)
+    if tuple == nil then
+        return {[DEFAULT_ORIGIN] = 0}
+    end
+    if tuple[PRIV_OPTS_FIELD_ID] == nil or
+       tuple[PRIV_OPTS_FIELD_ID].origins == nil then
+        return {[DEFAULT_ORIGIN] = tuple.privilege}
+    end
+    return tuple[PRIV_OPTS_FIELD_ID].origins
+end
+
+-- Return total privileges from the given origins.
+local function privilege_from_origins(origins)
+    local new_privilege = 0
+    for _, privilege in pairs(origins) do
+        new_privilege = bit.bor(new_privilege, privilege)
+    end
+    return new_privilege
+end
+
+-- Return retulting opts from the given origins.
+local function opts_from_origins(origins)
+    for name, privilege in pairs(origins) do
+        if privilege == 0 then
+            origins[name] = nil
```

Comment:
Better avoid modifying function arguments - this is confusing and error-prone.

Reply:
Thanks, fixed.